### PR TITLE
refactor: split action-reconciliation persistence tests into domain-focused modules (#594)

### LINKED ISSUE #594: refactor: split action-reconciliation persistence tests into domain-focused modules
## Summary
Split `control-plane/tests/test_service_persistence_action_reconciliation.py` into domain-focused modules so delegation, receipt binding, reconciliation, policy evaluation, and action-review surface coverage stop accumulating in one 5k+ line file.

The current action-reconciliation persistence suite is large enough that it raises review cost, slows failure triage, and makes targeted future additions harder than necessary. The tests already fall into recognizable families such as create-tracking-ticket soft write, shuffle delegation, isolated-executor delegation, reconciliation binding, policy evaluation, and review-surface rendering. Those families should become separate files or submodules with shared fixtures rather than one monolith.

## Scope
- split the current action-reconciliation persistence suite into smaller domain-focused test modules
- preserve existing fixture reuse and coverage depth while reducing file-level sprawl
- introduce shared helpers only where they reduce duplication without hiding test intent
- keep runtime behavior unchanged and avoid mixing production refactors into this issue
- keep unrelated test families out of scope

Part of: #592
Depends on: none
Parallelizable: Yes

## Execution order
2 of 6

## Acceptance criteria
- `test_service_persistence_action_reconciliation.py` is reduced substantially or replaced by a small coordinating shim
- the split modules reflect coherent behavior families rather than arbitrary line-count slicing
- focused and full test discovery continue to pass with equivalent coverage

## Verification
- run the affected action-reconciliation persistence tests directly and through repository test discovery
- manually review the resulting file layout for clear behavior-family boundaries

### PR DESCRIPTION
**Pull request by TommyKammy**
Closes #594
This PR was opened by codex-supervisor.
Latest Codex summary:

Split the action-reconciliation persistence suite into five domain-focused modules and replaced the original [test_service_persistence_action_reconciliation.py](/Users/jp.infra/Dev/AegisOps-worktrees/issue-594/control-plane/tests/test_service_persistence_action_reconciliation.py:1) with a compatibility shim that re-exports the split classes. The new modules are [create tracking ticket](/Users/jp.infra/Dev/AegisOps-worktrees/issue-594/control-plane/tests/test_service_persistence_action_reconciliation_create_tracking_ticket.py:1), [reconciliation](/Users/jp.infra/Dev/AegisOps-worktrees/issue-594/control-plane/tests/test_service_persistence_action_reconciliation_reconciliation.py:1), [delegation/policy](/Users/jp.infra/Dev/AegisOps-worktrees/issue-594/control-plane/tests/test_service_persistence_action_reconciliation_delegation.py:1), [reviewed requests](/Users/jp.infra/Dev/AegisOps-worktrees/issue-594/control-plane/tests/test_service_persistence_action_reconciliation_reviewed_requests.py:1), and [review surfaces](/Users/jp.infra/Dev/AegisOps-worktrees/issue-594/control-plane/tests/test_service_persistence_action_reconciliation_review_surfaces.py:1).

To keep discovery behavior stable, the split modules suppress direct discovery via empty `load_tests`, and the legacy shim remains the single discovery entrypoint for this family. I also updated the structural/regression checks in [test_service_persistence.py](/Users/jp.infra/Dev/AegisOps-worktrees/issue-594/control-plane/tests/...

## Summary by CodeRabbit

* **Tests**
  * Reorganized and expanded test coverage into multiple focused modules for delegation, persistence, reconciliation, review surfaces, and reviewed requests.
  * Added extensive end-to-end persistence and correlation tests covering success, failure modes, mismatches, validation errors, and edge cases.
  * Broadened test discovery/harnesses to scan the expanded split modules and expect the additional domain-focused test artifacts.
  * Test helpers updated to validate multiple referenced test artifacts and aggregate their contents for assertions.

### CODE REVIEW

**Comment by TommyKammy** [comment]
Tracked PR head `3abf588a303b6f011a4ab4b76d98840253f4167e` blocker cleared; progress has resumed.

- reason code: `cleared`
- current supervisor state: `waiting_ci`
- automatic retry: yes
- next action: continue the tracked PR workflow for PR #600 from the current active state.

File: control-plane/tests/test_service_persistence_action_reconciliation_create_tracking_ticket.py [hunk 1]
```diff
@@ -0,0 +1,1041 @@
...
+        )
+
+
+def load_tests(
+    loader: unittest.TestLoader,
+    tests: unittest.TestSuite,
+    pattern: str,
+) -> unittest.TestSuite:
+    del loader, pattern
+    return tests
```
[the remark this answers is not in the document]
Comment (TommyKammy) [reply control-plane/tests/test_service_persistence_action_reconciliation_create_tracking_ticket.py:1050]: The supervisor reprocessed this configured-bot finding on the current head `835b7c68066f5848b7450168dbf1f424d948706e` and classified it as stale.

Evidence: reviewer=coderabbitai file=control-plane/tests/test_service_persistence_action_reconciliation_create_tracking_ticket.py line=1050 processed_on_current_head=yes. Source: https://github.com/TommyKammy/AegisOps/pull/600#discussion_r3106680485

Under the configured `reply_and_resolve` policy, the supervisor is auto-resolving this stale thread now.

File: control-plane/tests/test_service_persistence_action_reconciliation_review_surfaces.py [hunk 2]
```diff
@@ -0,0 +1,1236 @@
...
+        self.assertEqual(inner_store.list(ActionRequestRecord), ())
+
+
+def load_tests(
+    loader: unittest.TestLoader,
+    tests: unittest.TestSuite,
+    pattern: str,
+) -> unittest.TestSuite:
+    del loader, pattern
+    return tests
```
[the remark this answers is not in the document]
Comment (TommyKammy) [reply control-plane/tests/test_service_persistence_action_reconciliation_review_surfaces.py:1236]: The supervisor reprocessed this configured-bot finding on the current head `835b7c68066f5848b7450168dbf1f424d948706e` and classified it as stale.

Evidence: reviewer=coderabbitai file=control-plane/tests/test_service_persistence_action_reconciliation_review_surfaces.py line=1236 processed_on_current_head=yes. Source: https://github.com/TommyKammy/AegisOps/pull/600#discussion_r3106680489

Under the configured `reply_and_resolve` policy, the supervisor is auto-resolving this stale thread now.
